src/instruments/src/Common/displayUnit.tsx

```diff
@@ -42,13 +43,15 @@ export const DisplayUnit: React.FC<DisplayUnitProps> = (props) => {
     });
 
     useEffect(() => {
-        if (state === DisplayUnitState.On && (potentiometer === 0 || electricityState === 0)) {
+        if (state !== DisplayUnitState.Off && props.failed) {
+            setState(DisplayUnitState.Off);
+        } else if (state === DisplayUnitState.On && (potentiometer === 0 || electricityState === 0)) {
             setState(DisplayUnitState.Standby);
             setTimer(10);
         } else if (state === DisplayUnitState.Standby && (potentiometer !== 0 && electricityState !== 0)) {
             setState(DisplayUnitState.On);
             setTimer(null);
-        } else if (state === DisplayUnitState.Off && (potentiometer !== 0 && electricityState !== 0)) {
+        } else if (state === DisplayUnitState.Off && (potentiometer !== 0 && electricityState !== 0 && !props.failed)) {
```

Comment:
I almost wonder if it's better to just declare a new state ("Failed") that is entered (and exited) upon failure and recovery and avoiding tagging on this condition for state OFF. It kind of breaks the flow of this hook which makes sense as controlling display standby/self-test/on timers, which already is a bit over-complex as is.

I think this would be a better foundation for more advanced failures involving displays, especially once we have to consider the display swapping.

Reply:
As discussed, this comment is more of a functional nature. I've added a few failures to show how the failure system works, but the actual individual failures will have to be built and/or improved in separate PRs.